x-pack/plugins/apm/server/routes/mobile/get_mobile_launches_by_location.ts

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  kqlQuery,
+  rangeQuery,
+  termQuery,
+} from '@kbn/observability-plugin/server';
+import { SERVICE_NAME } from '../../../common/es_fields/apm';
+import { APMEventClient } from '../../lib/helpers/create_es_client/create_apm_event_client';
+import { getOffsetInMs } from '../../../common/utils/get_offset_in_ms';
+import { getBucketSize } from '../../../common/utils/get_bucket_size';
+import { environmentQuery } from '../../../common/utils/environment_query';
+
+interface Props {
+  kuery: string;
+  apmEventClient: APMEventClient;
+  serviceName: string;
+  environment: string;
+  start: number;
+  end: number;
+  locationField?: string;
+  offset?: string;
+}
+
+export async function getLaunchesByLocation({
+  kuery,
+  apmEventClient,
+  serviceName,
+  environment,
+  start,
+  end,
+  locationField,
+  offset,
+}: Props) {
+  const { startWithOffset, endWithOffset } = getOffsetInMs({
+    start,
+    end,
+    offset,
+  });
+
+  const { intervalString } = getBucketSize({
+    start: startWithOffset,
+    end: endWithOffset,
+    minBucketSize: 60,
+  });
+
+  const aggs = {
+    launches: {
+      filter: {
+        terms: { ['labels.lifecycle_state']: ['created', 'active'] },
```

Comment:
nit: Can you please convert `labels.lifecycle_state` as constant. Similar https://github.com/LikeTheSalad/kibana/blob/3a48bc9289b7dc9b8ecdebbf224a55686a5078f1/x-pack/plugins/apm/common/es_fields/apm.ts#L192C1-L192C1  

Also, the values created and active can be enum.

Reply:
I've converted `labels.lifecycle_state` to constant 👍  However, regarding creating enums for `created` and `active`, I'm not sure what could be a proper file to place them? Also, I guess we'd have to create 2 separate enums, one for android events which would contain `created` and other for iOS's events with `active`?